File: api/models/support.cs
namespace Api.Model;

public class Support
{
  public Guid SupportId { get;set; }
  public string Name { get;set; } = string.Empty;
  public string Icon { get;set; } = "leaf";
  public bool IsMonthly { get;set; }
  public int Amount { get;set; }
  public List<string> Benefits { get;set; } = new List<string>();
  public DateTime CreateAt { get;set; }
  public Guid UserId { get;set; }
  public User? User { get;set; }
  public ICollection<Post> Post { get;set; } = new List<Post>();
}

File: web/src/lib/type/support.ts
import type { IconName } from "@/components/icons/icon";

// POST /api/support, GET /api/support/{id}/status, GET /api/support/creator/{userId}
export type Support = {
  supportId: string;
  name: string;
  icon: IconName;
  isMonthly: boolean;
  amount: number;
  benefits: string[];
  createAt: string;
  userId: string;
};

// POST /api/support (request)
export type SupportCreateRequest = {
  name: string;
  icon?: IconName | null;
  isMonthly: boolean;
  amount: number;
  benefits?: string[] | null;
};

// PATCH /api/support/{id} (request)
export type SupportPatchRequest = {
  name?: string | null;
  icon?: IconName | null;
  isMonthly?: boolean | null;
  amount?: number | null;
  benefits?: string[] | null;
};
